src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -388,6 +455,42 @@ private JsonRpcRequest ReadRequest(JToken json)
                 args is JArray argsArray ? (object)PartiallyParsePositionalArguments(argsArray) :
                 null;
 
+            // If method is $/progress, get the progress instance from the dictionary and call Report
+            string method = json.Value<string>("method");
+
+            if (string.Equals(method, ProgressRequestSpecialMethod, StringComparison.Ordinal))
+            {
+                try
+                {
+                    JToken progressId =
+                        args is JObject ? args["token"] :
+                        args is JArray ? args[0] :
+                        null;
+
+                    JToken value =
+                        args is JObject ? args["value"] :
+                        args is JArray ? args[1] :
+                        null;
+
+                    lock (this.progressMap)
+                    {
+                        //ProgressReceivedSignal.Reset();
+
+                        if (this.progressMap.TryGetValue(progressId.Value<long>(), out ProgressParamInformation progressInfo))
+                        {
+                            object typedValue = value.ToObject(progressInfo.valueType);
+                            progressInfo.reportMethod.Invoke(progressInfo.progressObject, new object[] { typedValue });
```

Comment:
>reportMethod [](start = 41, length = 12)

Do we want to guard against null here and report a more specific error message?

Reply:
I don't think is necessary, I'm adding the null check to make sure FindIProgressOfT() returns a valid IProgress<T> type, and we use that get the ReportMethod which I don't think there is chance of that being null. Maybe @andrewarnott@gmail.com can confirm.

---
In reply to: [304596198](https://github.com/milopezc/vs-streamjsonrpc/pull/1#discussion_r304596198) [](ancestors = 304596198)